crates/sargon/src/profile/mfa/secured_entity_control/secured_entity_control.rs

```diff
@@ -3,13 +3,26 @@ use crate::prelude::*;
 /// Advanced security control of an entity which has been "securified",
 /// meaning an MFA security structure (`SecurityStructureOfFactorSources`)
 /// which user has created has been applied to it.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash)]
+#[derive(
+    Serialize, Deserialize, Clone, PartialEq, Eq, Hash, derive_more::Debug,
+)]
 #[serde(rename_all = "camelCase")]
 pub struct SecuredEntityControl {
+    #[doc(hidden)]
+    #[serde(skip)]
+    #[debug(skip)]
+    pub __hidden: HiddenConstructor,
+
+    /// Virtual Entity Creation (Factor)Instance
+    ///
+    /// Optional since if we recovered this SecuredEntityControl part of
+    /// account recovery scan we might not know the veci
+    pub veci: Option<HierarchicalDeterministicFactorInstance>,
```

Comment:
This is new, needed to find VECI (Unsecurified keyspace) instances that **were** used by entities before they were securified. We NEVER want to reuse a FactorInstance in Unsecurified keyspace!

Reply:
Correct. But if we try to use that veci for a new entity, adding that entity will fail, since the entity address will be the same as the securities account. 

Effectively "re discovering" the veci.

So in future PRs we need to check add a check for save new virtual entity that if it fails, we check if an securified entity exists and then set the veci to the  newly rediscovered veci.

@GhenadieVP @matiasbzurovski I created JIRA ticket for this https://radixdlt.atlassian.net/browse/ABW-3988